server/datastore/mysql/migrations/tables/20230927155121_CreateTableQueryReports.go

```diff
@@ -0,0 +1,35 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20230927155121, Down_20230927155121)
+}
+
+func Up_20230927155121(tx *sql.Tx) error {
+	_, err := tx.Exec(`
+		CREATE TABLE query_results (
+			id INT UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+			query_id INT(10) UNSIGNED NOT NULL,
+			host_id INT(10) UNSIGNED NOT NULL,
+			osquery_version VARCHAR(50),
+			error TEXT COLLATE utf8mb4_unicode_ci DEFAULT NULL,
+			last_fetched TIMESTAMP NOT NULL,
+			data JSON,
+			FOREIGN KEY (query_id) REFERENCES queries(id) ON DELETE CASCADE,
+			FOREIGN KEY (host_id) REFERENCES hosts(id)
```

Comment:
See https://github.com/fleetdm/fleet/blob/fe0589c9e5109fdd65b56d79dd8330b0685f7d71/handbook/engineering/scaling-fleet.md#foreign-keys-and-locking

Let's not use a foreign key for `hosts`. It's the most "used" table and has brought issues in the past, and that's why we stopped adding foreign key references to it.

Reply:
Good catch @lucasmrod , note that it is important that tables that "would be foreign keys to hosts" are added to the `hostRefs` list here so they get deleted when the host is deleted: https://github.com/fleetdm/fleet/blob/71a237042a9c39a45bc8f9c76465e5ff6039eba9/server/datastore/mysql/hosts.go#L444 (I think that's where the link used to point in the doc you quoted, I'll update it).

Reply:
ah, nice catch.  I'll revert this from the feature branch and cut a new PR

Reply:
This feels like a good check to add to CI